### PR TITLE
Fixed some issues found by pvs

### PR DESCRIPTION
**Pull request by ProjectMutilation**

### CODE REVIEW

File: AI/Nullkiller/AIGateway.cpp [hunk 1]
```diff
@@ -710,8 +710,10 @@ void AIGateway::showGarrisonDialog(const CArmedInstance * up, const CGHeroInstan
 	//you can't request action from action-response thread
 	requestActionASAP([=]()
 	{
-		if(removableUnits && up->tempOwner == down->tempOwner)
+		if(removableUnits && up && down && up->tempOwner == down->tempOwner)
```
Comment (IvanSavenko) [inline AI/Nullkiller/AIGateway.cpp:713]: showGarrisonDialog is always called with non-null up & down. Othervice there is no point in such call - garrison dialog is about exchange between two armies.

I guess incorrect code is above this call:
```cpp
std::string s1 = up ? up->nodeName() : "NONE";
std::string s2 = down ? down->nodeName() : "NONE";
```
Check for nullptr is redundant in there.

Comment (ProjectMutilation) [reply AI/Nullkiller/AIGateway.cpp]: @IvanSavenko sorry, when I did rebase, I touched extra commits, but there should be no problems with merging
if it's problem can create PR with clear history

Comment (IvanSavenko) [reply AI/Nullkiller/AIGateway.cpp]: Why you can't simply drop them via interactive rebase?

File: Mods/vcmi/mod.json
Comment (Nordsoft91) [inline Mods/vcmi/mod.json:1]: unnecessary changes

File: AI/Nullkiller/AIGateway.cpp (hunk 1 above)
Comment (ProjectMutilation) [reply AI/Nullkiller/AIGateway.cpp]: > Why you can't simply drop them via interactive rebase?

fixed, I didn't know about the "drop" directive in rebase, thanks

**Review by IvanSavenko — APPROVED** [review]
